src/main/kotlin/app/revanced/patches/youtube/misc/downloadactions/HookDownloadActionsPatch.kt

```diff
@@ -0,0 +1,174 @@
+                    parentResult.classDef
+                )
+            }.resultOrThrow().let { setVisibilityMethod ->
+                setVisibilityMethod.mutableMethod.apply {
+                    // Find the index of if-nez
+                    val insertIndex = setVisibilityMethod.scanResult.patternScanResult!!.startIndex + 2
+                    // Get register values used in if-nez
+                    val insertRegister = getInstruction<OneRegisterInstruction>(insertIndex).registerA
+
+                    // Add instructions just above the index of if-nez
+                    addInstructions(
+                        insertIndex,
+                        """
+                        invoke-static {}, $INTEGRATIONS_DOWNLOAD_PLAYLIST_BUTTON_CLASS_DESCRIPTOR->isPlaylistDownloadButtonHooked()Z
+                        move-result v$insertRegister
+                        """
+                    )
+                }
+            }
+        }
+
+        // endregion
+
+        // region Hook Download Playlist Button OnClick method
+
+        DownloadPlaylistButtonOnClickFingerprint.resultOrThrow().let {
+            it.mutableMethod.apply {
+
+                // region Get the index of the instruction that initializes the onClickListener
+
+                val onClickListenerInitializeIndex = indexOfFirstInstructionOrThrow {
+                    val reference = ((this as? ReferenceInstruction)?.reference as? MethodReference)
+
+                    opcode == Opcode.INVOKE_VIRTUAL_RANGE
+                            && reference?.parameterTypes?.first() == "Ljava/lang/String;"
+                }
+
+                // endregion
+
+                // region Get the class that contains the onClick method
+
+                val onClickListenerInitializeReference =
+                    getInstruction<ReferenceInstruction>(onClickListenerInitializeIndex).reference
+
+
+                val onClickClass = context.findClass(
+                    (onClickListenerInitializeReference as MethodReference).returnType
+                )!!.mutableClass
+
+                // endregion
+
+                onClickClass.methods.find { method -> method.name == "onClick" }?.apply {
+
+                    // region Get the index of playlist id
+
+                    val insertIndex = implementation!!.instructions.indexOfFirst { instruction ->
+                        instruction.opcode == Opcode.INVOKE_STATIC
+                                && instruction.getReference<MethodReference>()?.name == "isEmpty"
+                    }
+
+                    val insertRegister = getInstruction<Instruction35c>(insertIndex).registerC
+
+                    // endregion
+
+                    addInstructions(
+                        insertIndex,
+                        """
+                        invoke-static {v$insertRegister}, $INTEGRATIONS_DOWNLOAD_PLAYLIST_BUTTON_CLASS_DESCRIPTOR->startPlaylistDownloadActivity(Ljava/lang/String;)Ljava/lang/String;
```

Comment:
The name of the method that hooks the video download button is `inAppDownloadButtonOnClick`, but the name of the method that hooks the playlist download button is `startPlaylistDownloadActivity`.

It is better to change the name of the method to a more similar structure.